libtenzir/src/pipeline.cpp

```diff
@@ -207,14 +217,60 @@ auto pipeline::optimize(expression const& filter, event_order order) const
       result.push_back(std::move(ops[0]));
       current_filter = trivially_true_expression();
     }
+    // do_not_optimize is the blocker
+    // should we have a seperate block optimization
+    // order_invarient: should it block optimization? I don't think so, should
+    // pass up original
+    if (opt.selection) {
+      if (opt.selection->selection_finished) {
+        current_selection = columnar_selection::no_columnar_selection();
+      } else {
+        current_selection = std::move(*opt.selection);
+      }
+      if (opt.selection->do_not_optimize_selection
+          && opt.selection->fields_of_interest) { // is blocking
```

Comment:
nullopt and empty vector currently mean the same thing, and they really shouldn't.